cmd/launcher/launcher.go

```diff
@@ -235,25 +176,143 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 		launcherUpdater, err := updater.NewUpdater(
 			ctx,
 			launcherPath,
-			updater.UpdateFinalizer(logger, runnerShutdown),
+			updater.UpdateFinalizer(logger, func() error {
+				// reboot systray on auto update
+				if systrayRunner != nil {
+					systrayRunner.Interrupt(nil)
+				}
+				return runnerShutdown()
+			}),
 			launcherUpdaterconfig,
 		)
 		if err != nil {
-			return errors.Wrap(err, "create launcher updater")
+			return fmt.Errorf("creating launcher updater: %w", err)
 		}
 		runGroup.Add(launcherUpdater.Execute, launcherUpdater.Interrupt)
 	}
 
+	err = runGroup.Run()
+	return fmt.Errorf("running service: %w", err)
+}
+
+func writePidFile(path string) error {
+	err := ioutil.WriteFile(path, []byte(strconv.Itoa(os.Getpid())), 0600)
+	if err != nil {
```

Comment:
1. These lines can be combined.
2. `ioutil.WriteFile` is deprecated, and should be (I think) `os.WriteFile`

And in doing that, the function is no shorter than the leaving it unsplit. But I guess there's some value in having it be reasonable self documenting. Okay.